machinist/mserver/factory.go

```diff
@@ -35,11 +33,36 @@ func DnsPort(dnsPort int) ControllerModifier {
 
 func WithKDnsFlags(mods ...kdns.DNSModifier) ControllerModifier {
 	return func(controller *Controller) error {
-		for _, m :=  range mods {
-			if err := m(controller.dnsServer); err != nil {
+		if controller.dnsServer == nil {
+			dnsServ, err := kdns.NewDNS(mods...)
+			if err != nil {
 				return err
 			}
+			controller.dnsServer = dnsServ
+		}
+		return nil
+	}
+}
+
+func WithStateFile(filepath string) ControllerModifier {
+	return func(controller *Controller) error {
+		s, err := state.ReadInController(filepath)
+		if err != nil {
```

Comment:
let's say the file on disk is corrupted (eg, SIGTERM, CTL+C, crash, ... in the middle of a write).
Should we log the error, and continue? Rather than keep the DNS down until an operator manually fixes the file?

(eg, if this happened at 4am, ignoring and continuing would basically allow the system to recover on its own, rather than require someone to wake up and fix it)

Reply:
yeah seems fine to me.